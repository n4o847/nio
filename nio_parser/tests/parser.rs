#[test]
fn test_parse() {
    use std::fs;

    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
        insta::glob!("inputs/*.nio", |path| {
            let input = fs::read_to_string(path).unwrap();
            let result = nio_parser::parse(&input);
            insta::assert_debug_snapshot!(&result);
        });
    });
}
