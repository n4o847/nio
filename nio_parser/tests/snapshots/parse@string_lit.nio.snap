---
source: nio_parser/tests/parser.rs
expression: "&result"
input_file: nio_parser/tests/inputs/string_lit.nio
---
Ok(
    Program {
        statements: [
            Expr(
                StringLit(
                    "hello",
                ),
            ),
        ],
    },
)
